command-handler/src/util/aws/aws-server.js

```diff
@@ -0,0 +1,397 @@
+
+        let maxRetries = 20;
+        let attempts;
+
+        for (attempts = 1; attempts <= maxRetries; attempts++) {
+            //wait 30 seconds
+            await delay(1000 * 30);
+
+            const server = await getInstance({ serverName });
+
+            if (server[0].State.Name === 'running') {
+            break;
+            } else {
+            log.info(`Attempt ${attempts} Failed. Backing off for 30 seconds`);
+            }
+            break;
+        }
+
+        if (attempts === maxRetries) {
+            try {
+            throw new Error(`Failed to initialize server in aws after ${attempts} retries`);
+            } catch (error) {
+                log.error({message: error.message, stack: error.stack});
+                app.client.chat.postEphemeral({
+                channel: `${body.channel.id}`,
+                user: `${body.user.id}`,
+                text: `Failed to initialize server in aws`
+                });
+                return;
+            } 
+        }
+
+        maxRetries = 4;  
+        for (attempts = 1; attempts <= maxRetries; attempts++) {
+            //wait 30 seconds
+            await delay(1000 * 30);
+            try {
+            //get servers and info from tailscale
+            const { deviceId } = await getDevices(serverName);
+    
+            await axios.post(`https://api.tailscale.com/api/v2/device/${deviceId}/tags`, 
+                {
+                "tags": [
+                    `tag:${userEmail}`
+                ]
+                }, {
+                headers: {
+                'Authorization': `Bearer ${process.env.TAILSCALE_API_TOKEN}`,
+                'Content-Type': 'application/json'
+                }
+            });
+            break;
+            } catch (error) {
+                log.info(`Attempt ${attempts} Failed. Backing off for 30 seconds`)
+            }
+        }
+
+        if (attempts === maxRetries) {
+            try {
+            throw new Error(`Failed to set tags in tailscale after ${attempts} retries`);
+            } catch (error) {
+                log.error({message: error.message, stack: error.stack});
+                app.client.chat.postEphemeral({
+                channel: `${body.channel.id}`,
+                user: `${body.user.id}`,
+                text: `Failed to set tags in tailscale`
+                });
+                return;
+            }
+        }
+
+        //get servers and info from tailscale
+        const { deviceIP } = await getDevices(serverName);
+
+        //return info for tailscale
+        app.client.chat.postEphemeral({
+            channel: `${body.channel.id}`,
+            user: `${body.user.id}`,
+            text: `The server has been created: https://login.tailscale.com/admin/machines/${deviceIP}`
+        });
+    },
+
+    deleteServer: async ({ app, body, instanceId, serverName }) => {
+        //get servers from tailscale
+        const { deviceId } = await getDevices(serverName)
```

Comment:
Should we have a util folder for tailscale? Looks like this is duplicated logic that we would also have in hetzner?